app/services/analytics.ts

```diff
@@ -0,0 +1,100 @@
+import { action } from '@ember-decorators/object';
+import { service } from '@ember-decorators/service';
+import { get } from '@ember/object';
+import Service from '@ember/service';
+import { task, waitForQueue } from 'ember-concurrency';
+import config from 'ember-get-config';
+
+export enum analyticPrivacy {
+    public = 'public',
+    private = 'private',
+    undefined = 'n/a',
+}
+
+export default class Analytics extends Service {
+    @service metrics;
+    @service session;
+    @service router;
+
+    // tslint:disable-next-line:variable-name
```

Comment:
This should be renamed to something like `trackPageTask` to avoid disabling the linter. You can make it a private method to make it explicit if you want.